wordpress-mobile/WordPress-Android
# Issue/12028 save for later

Parent issue #12028

This tasks extracts logic related to "save for later/bookmark" action on reader post list items. ReaderPostBookmarkUseCase contains all the related logic. It propagates events to the view layer using LiveData.

Merge Instructions
1. Review this PR
2. Make sure https://github.com/wordpress-mobile/WordPress-Android/pull/12358 is merged
3. Update target branch to develop
4. Remove "Not Ready for Merge" label
5. Merge this PR


To test:

Dialog
1. Clear app data
2. Open Reader - Following tab
3. Click on Bookmark action on one of the items
4. Notice a dialog is shown
5. Click on OK and go to Saved tab
6. Notice the post is saved there

Snackbar
1. Open Reader - Following tab
1. Click on Bookmark action on another item
2. Notice Snackbar is shown
3. Click on "View All" action in the snackbar
4. Notice the "Saved" tab is shown and the post is there

Offline
1. Open Reader - Following tab
1. Click on Bookmark action on yet another item
2. Notice snackbar is shown
3. Wait for cca 10s
4. Turn on airplane mode
5. Go to Saved tab
6. Open the post and notice the content (even images) are loaded correctly even though you are offline

Unbookmark
1. Click on unbookmark action on one of the saved items
2. Notice the post gets unbookmarked and is removed from the saved tab (it might still be there but in a collapsed mode with "Undo" action)


The new Discover tab can be potentially tested as well but it's a bit complicated. We'll test all the actions before it's released. I personally wouldn't worry about it now.

PR submission checklist:
